anomalib/data/base/video.py

```diff
@@ -74,9 +78,15 @@ def __getitem__(self, index: int) -> Dict[str, Union[str, Tensor]]:
                 self.pre_process(image=frame.numpy(), mask=mask) for frame, mask in zip(item["image"], item["mask"])
             ]
             item["image"] = torch.stack([item["image"] for item in processed_frames]).squeeze(0)
-            mask = item["mask"]
+            mask = Tensor(item["mask"])
             item["mask"] = torch.stack([item["mask"] for item in processed_frames]).squeeze(0)
             item["label"] = Tensor([1 in frame for frame in mask]).int().squeeze(0)
+            if self.task == TaskType.DETECTION:
+                item["boxes"] = [
+                    torch.empty((0, 4)) if frame.max() == 0 else masks_to_boxes(frame)
+                    for frame in item["mask"].view((-1, 1) + item["mask"].shape[-2:])
+                ]
```

Comment:
This line is a bit difficult to follow. Would it be possible to have a specific function for this line so the reader could go over it line by line for clarity?

Reply:
It turns out this was unnecessarily complicated and could be reduced to a single assignment.